src/Microsoft.AspNet.OData.Shared/PathItem.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.AspNet.OData
+{
+    /// <summary>
+    /// Class to hold the PathItem Name and its keyproperties
+    /// </summary>
+    public class PathItem
+    {
+        /// <summary>
+        /// Name of the Path Item (eg: entity name, entity set name)
+        /// </summary>
+        public string Name { internal set; get; }
+
+        /// <summary>
+        /// Whether the PathItem is a cast segment
+        /// </summary>
+        public bool IsCastType { internal set; get; }
```

Comment:
>IsCastType [](http://example.com/codeflow?start=20&length=10)

Maybe have a PathItemType that is "castSegment", "propertySegment", "navigationPropertySegment"